usr/share/rear/restore/BAREOS/default/400_restore_backup.sh

```diff
@@ -1,178 +1,114 @@
-        fi
-
-        echo "restore client=$BAREOS_CLIENT $RESTOREJOB $FILESET where=$TARGET_FS_ROOT select all done
-
-" |     bconsole
-
-        # wait for job to start
-        LogPrint "waiting for job to start"
-        while true
-        do
-            sleep 3
-            echo "status client=$BAREOS_CLIENT" | bconsole | grep -E "^JobId.* running." && break
-        done
-
-        # wait for job to finish
-        LogPrint "waiting for job to finish"
-        while true
-        do
-            sleep 10
-            echo "status client=$BAREOS_CLIENT" | bconsole | grep -E "^No Jobs running" >/dev/null && break
-        done
-        LogPrint "Restore job finished."
-    else
-
-    # Prompt the user that the system recreation has been done and that
-    # bconsole is about to be started.
-        LogPrint "
-The system is now ready for a restore via Bareos. bconsole will be started for
-you to restore the required files. It's assumed that you know what is necessary
-to restore - typically it will be a full backup.
-
-Do not exit 'bconsole' until all files are restored
-
-WARNING: The new root is mounted under '$TARGET_FS_ROOT'.
-
-Press ENTER to start bconsole"
-        read
-
-        bconsole
+    # bareos_recovery_mode == automatic: restore most recent backup automatically
+    
+    LogPrint "starting restore using bconsole:"
+    LogPrint "$RESTORE_CMD"
+
+    # example output of 'bcommand_json "$RESTORE_CMD"':
+    # {
+    #   "jsonrpc": "2.0",
+    #   "id": null,
+    #   "result": {
+    #     "query": [
+    #       {
+    #         "jobid": "18",
+    #         "level": "F",
+    #         "jobfiles": "59653",
+    #         "jobbytes": "2504970763",
+    #         "starttime": "2024-06-05 13:58:31",
+    #         "volumename": "Full-0001"
+    #       },
+    #        {
+    #         "jobid": "90",
+    #         "level": "I",
+    #         "jobfiles": "6553",
+    #         "jobbytes": "284336784",
+    #         "starttime": "2024-06-18 14:08:40",
+    #         "volumename": "Incremental-0009"
+    #       }
+    #     ],
+    #     "run": {
+    #       "jobid": "103"
+    #     }
+    #   }
+    # }
+    local restore_cmd_result
+    restore_cmd_result="$( bcommand_json "$RESTORE_CMD" )"
+    Log "$restore_cmd_result"
+    if [ -z "$restore_cmd_result" ] || ! restore_jobid="$( jq  --exit-status --raw-output '.result.run.jobid' <<< "$restore_cmd_result" )"; then
+        Error "Failed to determine Restore Job."
     fi
-    LogPrint "
-Please verify that the backup has been restored correctly to '$TARGET_FS_ROOT'
-in the provided shell. When finished, type exit in the shell to continue
-recovery.
-"
+    LogPrint "JobId of restore job is $restore_jobid"
+fi
 
-if [[ "$ISO_RECOVER_MODE" = "unattended" ]] || [[ "$PXE_RECOVER_MODE" = "unattended" ]] ; then
-    Log "Unattended mode selected"
+wait_restore_job "$restore_jobid"
+local job_exitcode=$?
+if [ ${job_exitcode} -eq 0 ]; then
```

Comment:
Just an observation: In some other places you used `(( ))` instead?

Reply:
adapted